crates/node/builder/src/builder/add_ons.rs

```diff
@@ -0,0 +1,71 @@
+//! Node add-ons. Depend on core [`NodeComponents`](crate::NodeComponents).
+
+use std::sync::Arc;
+
+use auto_impl::auto_impl;
+use reth_node_api::{FullNodeComponents, NodeAddOns};
+
+use crate::{
+    exex::BoxedLaunchExEx,
+    hooks::NodeHooks,
+    rpc::{EthApiBuilder, RpcHooks},
+};
+
+/// Additional node extensions.
+///
+/// At this point we consider all necessary components defined.
+pub struct AddOns<Node: FullNodeComponents, AddOns: NodeAddOns<Node>> {
```

Comment:
do we need to enforce this nodeaddons trait here?

can we not leave this up to the `Node` or the user entirely?